### ISSUE TITLE
Certain MODE messages could access on undefined

### ISSUE THREAD

**Issue by Acubed** [martynsmith/node-irc#144]
Presently I'm looking at https://github.com/martynsmith/node-irc/blob/35ee099b0517f47c51577403973b7416d1d29201/lib/irc.js#L229-L239 which does not verify that "user" is actually a user present in the channel.

This has occasionally crashed when a non-user value is passed as an argument:

```
[01 06:44:12] *** Mode #defocus +q *!*@adsl-76-237-180-172.dsl.chcgil.sbcglobal.net by mrmist
 TypeError: Cannot call method 'indexOf' of undefined
     at irc/lib/irc.js:233:54
```

Usually it works fine though, and I can't quite figure out why. But examining the code, almost certainly the server is capable of crafting a response that will throw a TypeError, disconnecting the IRC connection, and this should never happen.

On a similar note, the debug should probably look more like so:
```javascript
                if ( self.opt.debug )
                    util.log("MODE:" + message.args[0] + " sets mode: " + message.args.slice(1).join(' '));
```

**Comment by Jnull** [comment]
This can be closed.

**Comment by jirwin** [comment]
Fixed in #351.
